core/src/main/java/org/owasp/dependencycheck/analyzer/YarnClassicAuditAnalyzer.java

```diff
@@ -0,0 +1,202 @@
+ * limitations under the License.
+ *
+ * Copyright (c) 2021 The OWASP Foundation. All Rights Reserved.
+ */
+package org.owasp.dependencycheck.analyzer;
+
+import org.apache.commons.collections4.MultiValuedMap;
+import org.apache.commons.collections4.multimap.HashSetValuedHashMap;
+import org.apache.commons.io.IOUtils;
+import org.owasp.dependencycheck.Engine;
+import org.owasp.dependencycheck.analyzer.exception.AnalysisException;
+import org.owasp.dependencycheck.analyzer.exception.SearchException;
+import org.owasp.dependencycheck.analyzer.exception.UnexpectedAnalysisException;
+import org.owasp.dependencycheck.data.nodeaudit.Advisory;
+import org.owasp.dependencycheck.data.nodeaudit.NpmPayloadBuilder;
+import org.owasp.dependencycheck.dependency.Dependency;
+import org.owasp.dependencycheck.exception.InitializationException;
+import org.owasp.dependencycheck.utils.Settings;
+import org.owasp.dependencycheck.utils.URLConnectionFailureException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import us.springett.parsers.cpe.exceptions.CpeValidationException;
+
+import javax.annotation.concurrent.ThreadSafe;
+import jakarta.json.Json;
+import jakarta.json.JsonException;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonReader;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@ThreadSafe
+public class YarnClassicAuditAnalyzer extends AbstractYarnAuditAnalyzer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(YarnClassicAuditAnalyzer.class);
+
+    @Override
+    public String getName() {
+        return "Yarn Classic Audit Analyzer";
+    }
+
+    /**
+     * An expected error from `yarn audit --offline --verbose --json` that will
+     * be ignored.
+     */
+    private static final String EXPECTED_ERROR = "{\"type\":\"error\",\"data\":\"Can't make a request in "
+            + "offline mode (\\\"https://registry.yarnpkg.com/-/npm/v1/security/audits\\\")\"}\n";
+
+    /**
+     * Initializes the analyzer once before any analysis is performed.
+     *
+     * @param engine a reference to the dependency-check engine
+     * @throws InitializationException if there's an error during initialization
+     */
+    @Override
+    protected void prepareFileTypeAnalyzer(Engine engine) throws InitializationException {
+        super.prepareFileTypeAnalyzer(engine);
+        if (!isEnabled()) {
+            LOGGER.debug("{} Analyzer is disabled skipping yarn major version check", getName());
+            return;
+        }
+        var yarnMajorVersion = getYarnMajorVersion();
+
+        if (YARN_CLASSIC_MAJOR_VERSION == yarnMajorVersion) {
+            this.setEnabled(false);
+            LOGGER.debug("The {} has been disabled. This analyzer is not applicable for major version {}.", getName(), yarnMajorVersion);
+        }
```

Comment:
This was probably my error when moving from `String` -> `int` - I think the comparison was wrong. In 'Classic' mode we want to disable if the yarn Major version is greater than the classic major version.

```suggestion
        if (YARN_CLASSIC_MAJOR_VERSION > yarnMajorVersion) {
            this.setEnabled(false);
            LOGGER.debug("The {} has been disabled. This analyzer is not applicable for major version {}.", getName(), yarnMajorVersion);
        }
```

Reply:
Yes, good catch. I also should have noticed. I think you have switched berry and classic with your suggestions but I think I got it right now with the PR.